### PR TITLE
Combine ecflow and rocoto workflow generation

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**

This PR:
- consolidates the workflow generation systems for ecflow and rocoto.  Further unification will be made to make a choice to use one or the other.  At this time, they are separate and need separate input criteria.
- Move `ush/rocoto` to `workflow/` at the top level.  Further refining will be performed as needed based on use and need.

**Type of change**

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

**How Has This Been Tested?**
Since only the creation of experiment directory and XML are impacted, an experiment and XML was created from `develop`.  An experiment and XML from this PR was also created.  The differences were examined and were found to be as expected.
This test was performed on Hera.  The two experiment directories are at:
```
/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/scratch/EXPDIR/testdev
/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/scratch/EXPDIR/testcan
```
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] New and existing tests pass with my changes
- [X] Any dependent changes have been merged and published

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ __pycache__
 *.[aox]
 *.mod
 *.sw[a-p]
+.DS_Store
```
Comment (WalterKolczynski-NOAA) [inline .gitignore:8]: Where are these two hidden directories coming from?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply .gitignore:8]: (the second one being .idea, which was added in the last PR)

Comment (aerorahul) [reply .gitignore:8]: .DS_Store is created if you open the folder on a macOS Finder

.idea/ is a folder that is specific to PyCharm. It is an IDE that stores user configuration for the IDE; like interpreter, virtual env. etc.

Comment (WalterKolczynski-NOAA) [reply .gitignore:8]: Got it

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Will merge Tuesday.

**Pull request merged**